File: CozyHavenStayServer/CozyHavenStayServer/Models/DTO/SearchHotelDTO.cs
namespace CozyHavenStayServer.Models.DTO
{
    public class SearchHotelDTO
    {
        public string? Location { get; set; }
        public DateTime? CheckInDate { get; set; }
        public DateTime? CheckOutDate { get; set; }
        public int? NumberOfRooms { get; set; }
        public int? NumberOfAdults { get; set; }
        public int? NumberOfChildren { get; set; }
    }
}

File: CozyHeavenStay/src/app/models/DTO/searchHotelDTO.ts
export class Booking {
    Location: string | null;
    CheckInDate: Date | null;
    CheckOutDate: Date | null;
    NumberOfRooms: number | null;
    NumberOfAdults: number | null;
    NumberOfChildren: number | null;
  
    constructor(
      Location: string | null,
      CheckInDate: Date | null,
      CheckOutDate: Date | null,
      NumberOfRooms: number | null,
      NumberOfAdults: number | null,
      NumberOfChildren: number | null
    ) {
      this.Location = Location;
      this.CheckInDate = CheckInDate;
      this.CheckOutDate = CheckOutDate;
      this.NumberOfRooms = NumberOfRooms;
      this.NumberOfAdults = NumberOfAdults;
      this.NumberOfChildren = NumberOfChildren;
    }
  }
  